AISKU/snippet/snippet.js

```diff
@@ -1,106 +1,304 @@
+
+            return date.getUTCFullYear()
+                + "-" + pad(date.getUTCMonth() + 1)
+                + "-" + pad(date.getUTCDate())
+                + "T" + pad(date.getUTCHours())
+                + ":" + pad(date.getUTCMinutes())
+                + ":" + pad(date.getUTCSeconds())
+                + "." + String((date.getUTCMilliseconds() / 1000).toFixed(3)).slice(2, 5)
+                + "Z";            
+        }
+
+        function _createEnvelope(iKey, theType) {
+            var tags = {};
+            var type = "Browser";
+            tags[strAiDevice + "id"] = type[strToLowerCase]();
+            tags[strAiDevice + "type"] = type;
+            tags[strAiOperationName] = locn && locn.pathname || "_unknown_";
+            tags[strAiSdkVersion] = "javascript:snippet_" + (appInsights.snipVer || appInsights.version);
 
-    return appInsights;
-})({
-    instrumentationKey: "INSTRUMENTATION_KEY"
-   });
+            return {
+                time: _getTime(),
+                iKey: iKey,
+                name: "Microsoft.ApplicationInsights." + iKey.replace(/-/g, strEmpty) + "." + theType,
+                sampleRate: 100,
+                tags: tags,
+                data: {
+                    baseData: {
+                        ver: 2
+                    }
+                }
+            };
+        }
 
-// global instance must be set in this order to mitigate issues in ie8 and lower
-window[aiName] = aisdk;
+        function _createInternal(iKey, message, targetSrc) {
+            var envelope = _createEnvelope(iKey, "Message");
+            var data = envelope.data;
+            data.baseType = "MessageData";
+            var baseData = data.baseData;
+            baseData.message = "AI (Internal): 99 message:\"" + (message + " (" + targetSrc + ")").replace(/\"/g, strEmpty) + "\"";
+            baseData.properties = {};
 
-// if somebody calls the snippet twice, don't report page view again
-if (aisdk.queue && aisdk.queue.length === 0) {
-     aisdk.trackPageView({});
-}
+            return envelope;
+        }
+
+        function _createException(iKey, message, targetSrc) {
+            var envelope = _createEnvelope(iKey, "Exception");
+            var data = envelope.data;
+            data.baseType = "ExceptionData";
+            data.baseData.exceptions = [{
+                typeName: "ScriptLoadFailed",
+                message: message.replace(/\./g, "-"),  // Replacing '.' characters as it causes the portal to hide the start of the message in the summary
+                hasFullStack: false,
+                stack: message + "\nSnippet failed to load [" + targetSrc + "] -- Telemetry is disabled\nHelp Link: " + helpLink + "\nHost: " + (locn && locn.pathname || "_unknown_"),
+                parsedStack: []
+            }];
+
+            return envelope;
+        }
+    
+        // Assigning these to local variables allows them to be minified to save space:
+        var targetSrc = aiConfig.url || snipConfig.src;
```

Comment:
for the stuff other than what you are adding for handling snippet failures, are there any other behavior changes?

Reply:
No - Apart from "adding" the snipConfig (src, name, ld, useXhr -- see Readmd.md for documentation) which mostly are just exposing the previously internal only values -- so we just define them once